File: Interviews/LeetCode/Topic-Wise/Implementation/Easy/Determine-if-String-Halves-Are-Alike/1704.py
class Solution:
    def isVowel(self, c):
        return c == 'a' or c == 'e' or c == 'i' or c == 'o' or c == 'u'

    def halvesAreAlike(self, s: str) -> bool:
        cancelOut, i, j = 0, 0, len(s) - 1
        s = s.lower()
        while (i < j):
            cancelOut += 1 if self.isVowel(s[i]) else 0
            cancelOut -= 1 if self.isVowel(s[j]) else 0
            i, j = i + 1, j - 1

        return cancelOut == 0

File: Interviews/LeetCode/Daily-Problems/2022/December/Day-01/1704.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    bool isVowel(char c) {
        return c == 'a' or c == 'e' or c == 'i' or c == 'o' or c == 'u';
    }

    bool halvesAreAlike(string s) {
        int cancelOut = 0, i = 0, j = s.size() - 1;
        while (i < j) {
            cancelOut += isVowel(tolower(s[i++]));
            cancelOut -= isVowel(tolower(s[j--]));
        }

        return cancelOut == 0;
    }
};